examples/train_model.py

```diff
@@ -123,6 +124,11 @@ def __init__(self, parser):
             build_dict.build_dict(opt)
         # Create model and assign it to the specified task
         self.agent = create_agent(opt)
+        
+        # CUDA with multiprocessing requires 'spawn' start method
+        if opt.get('numthreads', 1) > 1 and opt.get('cuda', False):
+            set_start_method('spawn')
```

Comment:
I think we need to protect this call using `if __name__ == '__main__'`--so this should be in the training loop

Reply:
So at the bottom of the train file under the name/main check we should set multiprocessing to spawn if cuda is enabled (we could wrap that in a try/except checking if `import torch; if torch.is_cuda_enabled()` is true as well?)

Reply:
can't we move this to the world or something? it's annoying people have to put this in their train script isn't it?

Reply:
the multiprocessing docs say:
```
Note that this should be called at most once, and it should be protected inside the if __name__ == '__main__' clause of the main module.
```
it might work in HogwildWorld as well but this motivated me to put it in the main module!

Reply:
I tried putting it in HogwildWorld but was getting errors that the context was already created. Also valid world is also a HogWildWorld but the 'set_start_method('spawn')' call should appear only once in a program.

If in __main__ did you mean something like this:
 try:
        import torch
        from multiprocessing import set_start_method
        if torch.cuda.is_available():
            # CUDA with multiprocessing requires 'spawn' start method
            set_start_method('spawn')
    except:
        pass